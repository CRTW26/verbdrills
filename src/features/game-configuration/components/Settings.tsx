import React, { useState } from 'react'
import Button from 'shared/components/Button'
import { FaChevronUp, FaChevronDown } from 'react-icons/fa'

interface Props {
  defaultTime: number
}

const Settings: React.FC<Props> = ({ defaultTime }) => {
  const [time, setTime] = useState(defaultTime)

  return (
    <>
      <h1>Settings</h1>

      <div className="timer-setting">
        <h3>Timer</h3>

        <button onClick={() => setTime(time + 1)} disabled={time > 9}>
          <FaChevronUp color="#e0ff4f" fontSize={'2rem'} />
        </button>

        <p>
          {time} {`minute${time > 1 ? 's' : ''}`}
        </p>

        <button onClick={() => setTime(time - 1)} disabled={time < 2}>
          <FaChevronDown color="#e0ff4f" fontSize={'2rem'} />
        </button>
      </div>

      <Button className="btn btn--primary" text="Save" />

      <style jsx>{`
        .timer-setting {
          margin: 1rem 0;
          display: flex;
          flex-direction: column;
          align-items: center;
        }

        .timer-setting > h3 {
          align-self: flex-start;
        }

        button {
          background: none;
          border: none;
        }

        p {
          font-size: 1.5rem;
        }
      `}</style>
    </>
  )
}

export default Settings
