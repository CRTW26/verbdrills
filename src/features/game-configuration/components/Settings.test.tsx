import { fireEvent, render } from '@testing-library/react'
import Settings from './Settings'

describe('<Settings />', () => {
  test('should render', () => {
    const { baseElement } = render(
      <Settings defaultTime={1} onSave={jest.fn()} />
    )

    expect(baseElement).toMatchSnapshot()
  })

  test('should increment timer', () => {
    const { getByTestId } = render(
      <Settings defaultTime={1} onSave={jest.fn()} />
    )

    expect(getByTestId('time').textContent).toEqual('1 minute')

    fireEvent.click(getByTestId('button-increment'))

    expect(getByTestId('time').textContent).toEqual('2 minutes')
  })

  test('should decrement timer', () => {
    const { getByTestId } = render(
      <Settings defaultTime={2} onSave={jest.fn()} />
    )

    expect(getByTestId('time').textContent).toEqual('2 minutes')

    fireEvent.click(getByTestId('button-decrement'))

    expect(getByTestId('time').textContent).toEqual('1 minute')
  })
})
