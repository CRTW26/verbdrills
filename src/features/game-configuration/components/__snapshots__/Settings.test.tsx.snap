// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings /> should render 1`] = `
<DocumentFragment>
  <h1
    class="jsx-2491170599"
  >
    Settings
  </h1>
  <div
    class="jsx-2491170599 timer-setting"
  >
    <h3
      class="jsx-2491170599"
    >
      Timer
    </h3>
    <button
      class="jsx-2491170599"
      data-testid="button-increment"
    >
      <svg
        color="#e0ff4f"
        fill="currentColor"
        font-size="2rem"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(224, 255, 79);"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M240.971 130.524l194.343 194.343c9.373 9.373 9.373 24.569 0 33.941l-22.667 22.667c-9.357 9.357-24.522 9.375-33.901.04L224 227.495 69.255 381.516c-9.379 9.335-24.544 9.317-33.901-.04l-22.667-22.667c-9.373-9.373-9.373-24.569 0-33.941L207.03 130.525c9.372-9.373 24.568-9.373 33.941-.001z"
        />
      </svg>
    </button>
    <p
      class="jsx-2491170599"
      data-testid="time"
    >
      1 minute
    </p>
    <button
      class="jsx-2491170599"
      data-testid="button-decrement"
      disabled=""
    >
      <svg
        color="#e0ff4f"
        fill="currentColor"
        font-size="2rem"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(224, 255, 79);"
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        />
      </svg>
    </button>
  </div>
  <button
    class="jsx-767425152 btn btn--primary"
  >
    Save
  </button>
</DocumentFragment>
`;
